.github/workflows/freebsd.yml

```diff
@@ -0,0 +1,37 @@
+# GitHub Actions workflow to run tests on a FreeBSD VM.
+name: "FreeBSD"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *" # At 00:00 daily.
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.event.number || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "FreeBSD"
+    runs-on: ubuntu-latest
+    if: ${{ github.repository_owner == 'libressl' || github.event_name != 'schedule' }}
+    permissions:
+      contents: read
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+
+      - name: "Setup"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y automake autoconf libtool
+          ./autogen.sh
+
+      - name: "Build on VM"
+        uses: vmactions/freebsd-vm@v1
+        with:
+          prepare: |
+            pkg install -y gcc gmake
```

Comment:
I would not install gcc. This should be using FreeBSD's clang, which is what `cc` defaults to. There's no need to add `gmake`, but you'll need to `pkg install -y autoconf automake libtool` for the autoconf build and `pkg install cmake ninja` for the cmake build.